### ISSUE TITLE
access hidden_states of MambaLMHeadModel

### ISSUE THREAD

**Issue by s6mahahn** [state-spaces/mamba#337]
I pretrained a MambaLMHeadModel with custom data using next token prediction. Now I want to use that to embedd sequences and use these seqences to classify them. Unfortunatly there seems no way to access the hidden states of the model. I can only get the logits of the model. Is there a way to access them?

**Comment by BlenderWang9487** [comment]
I assume that the "hidden_states" you mentioned refer to the outputs of the last layer of the model before passing through the `lm_head`. You can create a simple custom forward function, like this:

(I copied the MambaLMHeadModel's forward function and made some modifications)

```python
def my_custom_forward(self, input_ids, position_ids=None, inference_params=None, num_last_tokens=0):
        """
        "position_ids" is just to be compatible with Transformer generation. We don't use it.
        num_last_tokens: if > 0, only return the logits for the last n tokens
        """
        hidden_states = self.backbone(input_ids, inference_params=inference_params)
        if num_last_tokens > 0:
            hidden_states = hidden_states[:, -num_last_tokens:]
        # lm_logits = self.lm_head(hidden_states)
        # CausalLMOutput = namedtuple("CausalLMOutput", ["logits"])
        # return CausalLMOutput(logits=lm_logits)
        
        # Directly return hidden_states
        return hidden_states
```

If you originally called the model like this:
```python
output = model(input_ids)
```

Now you can do:
```python
hidden_states = my_custom_forward(model, input_ids)
```

Is this what you want to do?
